### DISCUSSION TITLE
Using the same project auth on sub domain and main domain

### DISCUSSION THREAD

**Discussion by onurusluca** [supabase/supabase#30483] [Questions] [answered]
Hi, thanks a lot for this great library.

I have this problem I can't seem to figure out.
When I login on an account on my main domain, I am also logged out on my sub domain. 
Is there a way to keep the domains seperated? I don't want to use different projects as data is shared across as my website is multi-tenant.

Say I have this domain and sub domain: mywebsite.com and vendor.mywebsite.com. I want to be able to log in with different account on these websites. 

I have searched for long but couldn't find any reliable solutions. Any have any idea on how to handle this case?
Found this discussion: https://github.com/orgs/supabase/discussions/5742
But it looks that, using different domains (e.g., vendor.example.com and store.example.com) should actually keep the auth states separate by default?

Thanks.

**Comment by sadlrxn** [comment]
hello, if you happenened to find out how to do it, can you help me out?

**Reply by onurusluca** [reply, marked as answer]
> hello, if you happenened to find out how to do it, can you help me out?

Hey. I found out that this happens only on local, when you deploy it, it works fine. Supabase understands they are different domains I guess?

So I can log in to vendor.mysite.com and my site.com with different account on the same browser.

**Comment by sadlrxn** [comment]
I have an app that has to authenticate subdomains.  my configurations are:

```js
    cookieOptions: {
      domain: '.my-app.com',
      sameSite: 'lax',
      maxAge: 60 * 60 * 24,
      httpOnly: true,
      path: "/",
    },

```

I have proxy settings  my-app.com pointing to localhost.
I can't login and i can't see cokkies set in browser. 

![Screenshot from 2025-02-24 09-12-06](https://github.com/user-attachments/assets/8865d7dc-bf40-4ce1-9582-3a825e387e94)

that is my supabase url configuration
Can you help me out?


Actually, i want to authenticated sub domains and main domain at the same time.
forexample if a log in at my-app.com, i want to be logged in at `sub.my-app.com'

i am using nuxt.

**Reply by onurusluca** [reply]
Take a look at this
https://github.com/orgs/supabase/discussions/5742
